src/api_server/src/lib.rs

```diff
@@ -244,16 +246,25 @@ impl ApiServer {
         request: &Request,
         request_processing_start_us: u64,
     ) -> Response {
-        match ParsedRequest::try_from_request(request) {
-            Ok(ParsedRequest::Sync(vmm_action)) => {
-                self.serve_vmm_action_request(vmm_action, request_processing_start_us)
-            }
-            Ok(ParsedRequest::GetMMDS) => self.get_mmds(),
-            Ok(ParsedRequest::PatchMMDS(value)) => self.patch_mmds(value),
-            Ok(ParsedRequest::PutMMDS(value)) => self.put_mmds(value),
-            Ok(ParsedRequest::ShutdownInternal) => {
-                self.shutdown_flag = true;
-                Response::new(Version::Http11, StatusCode::NoContent)
+        match ParsedRequest::try_from_request(request).map(|r| r.into_parts()) {
+            Ok((req_action, mut parsing_info)) => {
+                let mut response = match req_action {
+                    RequestAction::Sync(vmm_action) => {
+                        self.serve_vmm_action_request(vmm_action, request_processing_start_us)
+                    }
+                    RequestAction::GetMMDS => self.get_mmds(),
+                    RequestAction::PatchMMDS(value) => self.patch_mmds(value),
+                    RequestAction::PutMMDS(value) => self.put_mmds(value),
+                    RequestAction::ShutdownInternal => {
+                        self.shutdown_flag = true;
+                        Response::new(Version::Http11, StatusCode::NoContent)
+                    }
+                };
+                if parsing_info.deprecated() {
+                    warn!("{}", parsing_info.take_deprecation_message().unwrap());
+                    response.set_deprecation();
+                }
```

Comment:
I would change the if to something like:
```
if let Some(deprecation_message) = parsing_info.take_deprecation_message() {
    ...
}
```

If we do this do we still need the `deprecated()` method ? If we do, I would rename it to `has_deprecation _message()`.

Reply:
It was intended for parts of the code that just want to check that a deprecation message was set without altering the state of the message by doing the `take`. As this was not useful right now, I removed the function and we can add it in the future if we need it.